Yank/schema/validator.py

```diff
@@ -102,10 +113,135 @@ def _validate_type_quantity(self, value):
         if isinstance(value, unit.Quantity):
             return True
 
+    # ====================================================
+    # INTERNAL USAGE
+    # ====================================================
+
+    def _check_subclass_constructor(self, field, call_constructor_func,
+                                    constructor_description):
+        """Utility function for logging constructor validation errors."""
+        try:
+            call_constructor_func(constructor_description)
+        except RuntimeError as e:
+            self._error(field, str(e))
+
+
+# ==============================================================================
+# STATIC VALIDATORS/COERCERS
+# ==============================================================================
+
+def to_unit_coercer(compatible_units):
+    """Function generator to test unit bearing strings with Cerberus."""
+    def _to_unit_coercer(quantity_str):
+        return utils.quantity_from_string(quantity_str, compatible_units)
+    return _to_unit_coercer
+
+
+def to_integer_or_infinity_coercer(value):
+    """Coerce everything that is not infinity to an integer."""
+    if value != float('inf'):
+        value = int(value)
+    return value
+
 
-# ====================================================
-# UTILITY FUNCTIONS
-# ====================================================
+# ==============================================================================
+# OBJECT CONSTRUCTORS PARSING
+# ==============================================================================
+
+def call_constructor(parent_cls, constructor_description, special_conversions=None,
```

Comment:
Ooohhhhh... I like this.... I'll need to spend more time wrapping my head exactly what it does, but I very much like this